cargo-shuttle/src/client.rs

```diff
@@ -174,8 +175,22 @@ impl Client {
         self.delete(path).await
     }
 
-    pub async fn get_logs(&self, project: &str, deployment_id: &Uuid) -> Result<Vec<LogItem>> {
-        let path = format!("/projects/{project}/deployments/{deployment_id}/logs");
+    pub async fn get_logs(
+        &self,
+        project: &str,
+        deployment_id: &Uuid,
+        mode: LogMode,
+        len: u32,
+    ) -> Result<Vec<LogItem>> {
+        let path = match mode {
+            LogMode::Head => {
+                format!("/projects/{project}/deployments/{deployment_id}/logs?mode=head&len={len}")
+            }
+            LogMode::Tail => {
+                format!("/projects/{project}/deployments/{deployment_id}/logs?mode=tail&len={len}")
+            }
+            LogMode::All => format!("/projects/{project}/deployments/{deployment_id}/logs"),
+        };
```

Comment:
The first part of the string is the same for all of them, so we don't need to repeat that for each variant. So we can have a url string at the start which is `/projects/{project}/deployments/{deployment_id}/logs`, and then if the mode is tail or head, we add the query params at the end.